crates/provider-sdk/src/lib.rs

```diff
@@ -229,26 +268,56 @@ pub trait Provider<E = anyhow::Error>: Sync {
     }
 
     /// Notify the provider that the link is dropped
+    #[deprecated(
+        since = "0.6.1",
+        note = "prefer delete_configured_link_as_target() or delete_configured_link_as_source()"
+    )]
     fn delete_link(&self, component_id: &str) -> impl Future<Output = Result<(), E>> + Send {
         let _ = component_id;
         async { Ok(()) }
     }
 
     /// Notify the provider that the link is dropped where the provider is the target
+    #[deprecated(
+        since = "0.6.1",
+        note = "prefer delete_configured_link_as_target() instead. In a future release this method will be replaced with the implementation of delete_configured_link_as_target()"
+    )]
     fn delete_link_as_target(
         &self,
-        component_id: &str,
+        _component_id: &str,
+    ) -> impl Future<Output = Result<(), E>> + Send {
+        async { Ok(()) }
+    }
+
+    /// Notify the provider that the link is dropped where the provider is the target
+    ///
+    /// Unlike [`delete_link_as_target`], metadata like link name are available to be inspected.
+    fn delete_configured_link_as_target(
+        &self,
+        _info: impl LinkDeleteInfo,
     ) -> impl Future<Output = Result<(), E>> + Send {
-        let _ = component_id;
         async { Ok(()) }
     }
 
     /// Notify the provider that the link is dropped where the provider is the source
+    #[deprecated(
+        since = "0.6.0",
+        note = "prefer delete_configured_link_as_source() instead. In a future release this method will be replaced with the implementation of delete_configured_link_as_source()"
+    )]
     fn delete_link_as_source(
         &self,
-        component_id: &str,
+        _component_id: &str,
+    ) -> impl Future<Output = Result<(), E>> + Send {
+        async { Ok(()) }
+    }
+
+    /// Notify the provider that the link is dropped where the provider is the source
+    ///
+    /// Unlike [`delete_link_as_source`], metadata like link name are available to be inspected.
+    fn delete_configured_link_as_source(
+        &self,
+        _info: impl LinkDeleteInfo,
```

Comment:
Alternatively to the above, could we simply proxy this function to `self.delete_link_as_source` so that if the developer doesn't implement the function, we just call the "old" method for now?

Reply:
Ah yes! That's what I should have done in the first one -- we can switch to using the new stuff in internal code if we use the existing calls when not specified

Reply:
On second thought, I think we should just make this a breaking change -- it doesn't affect the host but actually only affects `provider-sdk` -- what do you think?

Reply:
I think we can just make this a breaking change, let's just keep the reasonable names that we had before & we can add release notes with the new version! It's just better

Reply:
Yup I agree! definitely should just be a breaking change -- I updated all the providers